haystack/retriever/elasticsearch.py

```diff
@@ -9,13 +9,57 @@
 
 
 class ElasticsearchRetriever(BaseRetriever):
-    def __init__(self, document_store: Type[BaseDocumentStore], direct_filters=None, custom_query=None):
+    def __init__(self, document_store: Type[BaseDocumentStore], custom_query: str = None):
+        """
+        :param document_store: an instance of a DocumentStore to retrieve documents from.
+        :param custom_query: query string as per Elasticsearch DSL with a mandatory question placeholder($question).
+
+                             Optionally, ES `filter` clause can be added where the values of `terms` are placeholders
+                             that get substituted during runtime. The placeholder(${filter_name_1}, ${filter_name_2}..)
+                             names must match with the filters dict supplied in self.retrieve().
+
+                             An example custom_query:
+                             {
+                                "size": 10,
+                                "query": {
+                                    "bool": {
+                                        "should": [
+                                            {
+                                                "multi_match": {
+                                                    "query": "${question}",  // mandatory $question placeholder
+                                                    "type": "most_fields",
+                                                    "fields": [
+                                                        "text",
+                                                        "title"
+                                                    ]
+                                                }
+                                            }
+                                        ],
+                                        "filter": [
+                                            {
+                                                "terms": {
+                                                    "year": ${years}  // optional custom filter
```

Comment:
Shouldn't this be ${year} to be compliant with the example of self.retriever() below? Same for quarters